python/tskit/tables.py

```diff
@@ -1815,3 +1817,23 @@ def unpack_strings(packed, offset, encoding="utf8"):
     :rtype: list[str]
     """
     return [b.decode(encoding) for b in unpack_bytes(packed, offset)]
+
+
+def to_np_int32(int_array):
+    """
+    A few functions require arrays of type np.int32. To allow passing standard numpy
+    integer arrays (dtype=np.int64) we cast but check bounds to avoid wrap-around
+    conversion errors (numpy doesn't seem to provide this functionality)
+    """
+    int_array = np.array(int_array)
+    if int_array.size == 0:
+        return int_array.astype(np.int32)  # Allow empty arrays of any type
+    try:
+        return int_array.astype(np.int32, casting='safe')
+    except TypeError:
+        if (int_array >= np.iinfo(np.int32).min).all() and \
```

Comment:
Cleaner to get a reference to np.iinfo(np.int32) in a variable, wouldn't need line break then.

Reply:
Good point. Done